File: chicken_and_dog.pas
Program chicken_and_dog;
Uses Crt;
const numLeg1d = 4;
const numLeg1c = 2;
var dog_img, dog, chick, chick_img, sum_leg, sum_dc : integer;
Begin
dog_img := 0;
while dog_img <= 36 do 
Begin
chick_img := 0;
while chick_img <= 36 do
Begin
sum_dc := dog_img + chick_img;
sum_leg := dog_img * numLeg1d + chick_img * numLeg1c;
if (sum_leg = 100) and (sum_dc = 36) then
Begin
dog := dog_img;
chick := chick_img;
writeln('Chickens: ', chick);
writeln('Dogs: ', dog);
break;
End;
chick_img := chick_img + 1;
if (dog <> 0) and (chick <> 0) then break;
End;
dog_img := dog_img + 1;
End;
End.

File: chicken_and_dog.c
#include <stdio.h>
#include <string.h>

//Defining some constants
const int numLeg1d = 4;
const int numLeg1c = 2;

int main()
{
    int dog = 0, chick = 0, sum_dc, sum_leg;
    for(int dog_img = 0; dog_img <= 36; dog_img++)
    {
        for(int chick_img = 0; chick_img <= 36; chick_img++)
        {
            sum_dc = dog_img + chick_img;
            sum_leg = dog_img * numLeg1d + chick_img * numLeg1c;
            if((sum_dc == 36) && (sum_leg == 100))
            {
                dog = dog_img, chick = chick_img;
                printf("%s", "Chickens: "); printf("%i", chick); printf("%c", '\n');
                printf("%s", "Dogs: "); printf("%i", dog); printf("%c", '\n');
                break;
            }
        }
        if(dog != 0 && chick != 0)
        {
            break;
        }
    }

    return 0;
}
